code/app/HTTP/Controllers/LodestoneController.php

```diff
@@ -20,74 +20,78 @@ public function __construct(API $lodestone)
         $this->lodestone = $lodestone;
     }
 
+    public function findCharacters(Request $request)
+    {
+        $this->validate($request, [
+            'name' => 'required|string',
+            'worldname' => 'sometimes|exists:worlds,name',
+            'page' => 'sometimes|int',
+        ]);
+
+        $results = $this->lodestone->findCharacters(
+            $request->name,
+            $request->worldname,
+            $request->page
+        );
+
+        if (!$results){
+            abort(404);
+        }
+
+        return json_encode($results);
```

Comment:
I could be wrong but I think laravel automatically encodes to JSON when JSON is requested - if not, we probably want to use `return response()->json($results);`